test/native/test_runner_main.cpp

```diff
@@ -0,0 +1,11 @@
+/*
+ * Copyright The async-profiler authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "test_runner.hpp"
+
+int main()
```

Comment:
Redundant class, imo.
TestRunner is essentially an entry point, you may just add `main` there.